File: src/main/java/com/vc/medium/L442.scala
package com.vc.medium

object L442 {
  def findDuplicates(arr: Array[Int]): List[Int] = {
    import scala.collection.mutable
    val list = new mutable.ListBuffer[Int]()
    arr.indices.foreach(i => {
      val idx = Math.abs(arr(i)) - 1
      if(arr(idx) < 0) list += idx + 1
      else arr(idx) = -1 * arr(idx)
    })
    list.toList
  }
}

File: src/main/java/com/vc/medium/L442.java
package com.vc.medium;

import java.util.*;

class L442 {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> res = new ArrayList<>();
        for(int i = 0; i < nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            if(nums[index] < 0) res.add(index + 1);
            else nums[index] = -1 * nums[index];
        }
        return res;
    }
}
